static/form.js

```diff
@@ -10,34 +9,99 @@ $(document).ready(function () {
 var destinationCounter = 2;
 
 function addDestination() {
+	var additionalDestinationsDiv = document.getElementById('additionalDestinations');
 
-  var additionalDestinationsDiv = document.getElementById('additionalDestinations');
-
-    // Create a new destination row
+	// Create a new destination row if there are no rows or the last row is full
+	if (!additionalDestinationsDiv.lastChild || additionalDestinationsDiv.lastChild.childElementCount === 2) {
 		var destinationRow = document.createElement('div');
 		destinationRow.classList.add('row');
+        additionalDestinationsDiv.appendChild(destinationRow);
+	}
+
+	// Create Destination Location input
+	var destinationLocationInput = document.createElement('div');
```

Comment:
It seems you have reverted Jacob's compressing of each new destination, can you use it since I merged with it as he made it as well. It looks better and I have continued my work on an extra destination deletion button with it. Also you can use the empty space to the right to place the dropdown menu.